### ISSUE TITLE
AzureML | SDKv2 | MLClient | Pass proxy variables

### ISSUE THREAD

**Issue by IliasAarab** [Azure/azure-sdk-for-python#36381]
Is it possible to add proxy/ssl config to the `MLClient`? 

```{python}
from azure.ai.ml import MLClient
from azure.identity import DefaultAzureCredential

ml_client = MLClient(
subscription_id=subscription_id,
resource_group_name=resource_group,
credential=DefaultAzureCredential(),
)
```

**Comment by mccoyp** [comment]
Hi @IliasAarab, thank you for opening an issue. Proxy and pipeline configuration can be set in SDK clients through the mechanisms described in our Core [client library developer reference](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md) documentation.

For [proxies specifically](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md#proxy-settings), you can either use the `HTTP_PROXY` and `HTTPS_PROXY` environment variables or pass a `azure.core.pipeline.policies.ProxyPolicy` instance to your client constructor in a `proxy_policy` keyword argument.

**Comment by IliasAarab** [comment]
@mccoyp thank you for the guidance! I'm still having an issue when trying to initialize an `MLClient` instance when passing a credential object to the constructor: 

```{python}
from azure.ai.ml import MLClient
from azure.identity import InteractiveBrowserCredential
from azure.core.pipeline.policies import ProxyPolicy

proxies = {"http": "myproxy", "https":  "myproxy"}
proxy_policy = ProxyPolicy()
proxy_policy.proxies = proxies

credential = InteractiveBrowserCredential()
client = MLClient.from_config(
    credential=credential,
    proxy_policy=proxy_policy
)
client.compute.get("is-cpu")
```

```{bash}
InteractiveBrowserCredential.get_token failed: Authentication failed: <urllib3.connection.HTTPSConnection object at 0x000001F13531D810>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed

ClientAuthenticationError: Authentication failed: <urllib3.connection.HTTPSConnection object at 0x000001F13531D810>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed
```

Everything works fine when I set my proxies as global env variables. (but that's not a viable option for all use cases). 

When I first set global env variables for the initial interactive authentication, and then continue work with the client (having passed proxy_policy) with global env variables deleted, everything still works fine. So the current problem resides with the initial auth.

Also, wondering how to pass `requests_ca_bundle` to the ProxyPolicy or to disable ssl verification (e.g., `verify=False`.

---

/unresolve

**Comment by mccoyp** [comment]
@IliasAarab thank you for the information. @xiangyan99, would you be able to take a look?

**Comment by IliasAarab** [comment]
Hi @xiangyan99, any updates on this ?

**Comment by xiangyan99** [comment]
Could you help to clarify your request?

In your case, it failed because `InteractiveBrowserCredential` does not support proxy.

Do you ask to add proxy support for `InteractiveBrowserCredential`?

**Comment by IliasAarab** [comment]
@xiangyan99 yes I guess that's what it boils down to. Or alternatively, whether there is another `Credential` object similar to `InteractiveBrowserCredential` that supports proxies? Otherwise, I am confused as what the added value is of the `proxy_policy` arg into `MLClient` as the client also needs a `Credential` instance.

**Comment by IliasAarab** [comment]
@xiangyan99  let me know if there is an update please.

**Comment by IliasAarab** [comment]
Please let me know if there is any update, or a decision on some course of action.

**Comment by xiangyan99** [comment]
This separation of the credential object and the client object is intentional, and their proxy settings need to be configured independently.

The credential object also supports a `proxies` parameter.

To resolve this, ensure you specify the `proxies` setting when creating the credential object.

Feel free to let me know if this resolves the issue for you.

**Comment by IliasAarab** [comment]
@xiangyan99 Thank you for the response. Configuring both objects in the same way works as expected indeed based on some trial runs! Will try out more extensively soon, but will close the issue for now. 

Thanks for the help.
